### ISSUE TITLE
Gradient of determinant is inexplicably slow

### ISSUE THREAD

**Issue by dpfau** [google/jax#2510]
Running this on CPU (on Google hardware), the forward computation is fairly quick, but the gradient is bizarrely slow (like almost a minute):

```
foo = lambda x: jnp.sum(jax.vmap(jnp.linalg.det, (0,))(x))
data = np.random.randn(10, 5, 5)
print(foo(data)) # reasonable
print(jax.grad(foo)(data)) # slow
```

It seems slow on GPU as well, but either way it's bizarre that it would take so long for such a simple function.

**Comment by shoyer** [comment]
See https://github.com/google/jax/issues/2380 for some discussion of custom gradient rules for `det`/`slogdet`, which might offer some improvements.

**Comment by dpfau** [comment]
Given I wrote the gradient for the determinant op, and understand well the issues around computing it with singular matrices - I don't think these issues are related. I'm computing this gradient with a full-rank matrix, which should just have the computational overhead of doing np.linalg.det * np.linalg.inv. Should be very fast.

**Comment by shoyer** [comment]
When I test this in the public version of Colab performance seems to be pretty reasonable: 
https://colab.research.google.com/gist/shoyer/ea906f210209a7998c72c0bd500646c5/grad-det-bench.ipynb

Both lines run in less than a millisecond after `jit` compilation.

This suggests it might be something related to our configuration at Google.

**Comment by dpfau** [comment]
Very possible. Matt just suggested I post it on GitHub.

**Comment by dpfau** [comment]
I'm saying it's slow *without* jit

**Comment by dpfau** [comment]
Seemed to be an issue with the Google internal configuration.
